src/content/en/2021/jamstack.md

```diff
@@ -1,20 +1,521 @@
+2. In our analysis, we can't get any info related to headless CMSs, hence we will not cover this either.
+3. We visualise SSG data using top 5 used SSGs based on number of sites built with these SSGs.
+
+Most of the data is represented for mobile and desktop websites. There's a fully described [methodology](./methodology) used in HTTPArchive.
+
+## Adoption of SSGs
+
+SSG adoption year over year in general is growing 2x times. In 2019 it was just 0.4% mobile and 0.3% desktop sites. In 2020 the number almost doubled, to 0.6% on mobile and 0.7% on desktop sites. In 2021 they have grown again: 1.1% of mobile and 0.9% of desktop sites.
+That underlines the trend of that technology. For example, this year a company like Vercel raised a $104M series C round and $150M in round D of investment to build a better web with modern technologies like Next.js. Also Jamstack oriented CDN provider Netlify raised $105M in their series D of investment. Hence, it's expected that numbers of Jamstack adoption will grow even higher next year.
+
+{{ figure_markup(
+  image="year-over-year-adoption.png",
+  caption="SSG adoption year over year",
+  description="Bar chart represents adoption year over year. Adoption year over year in general is growing 2x times.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=1588757197&format=interactive",
+  sheets_gid="874607830",
+  sql_file="ssg_compared_to_2020.sql"
+) }}
+
+In 2020 the amount of desktop websites increased 2.76 times, while mobile just 1.5 times. In 2021 mobile availability for SSGs built sites became way better than in 2020, and this year there are ~1.9 times more sites than 2020.
+
+## Which SSGs are the most popular
+
+{{ figure_markup(
+  image="adoption-share.png",
+  caption="SSG adoption share",
+  description="Pie chart represents adoption share between SSGs. Nuxt.js covers 52.6% of Jamstack sites. Next.js is in second place with 36.8%, third one is Gatsby with 6.7%, followed by Hugo at 2.5%.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRUiAdcUCSYyS7cQcQlE9uKeMSacgRudtf4pFRkIKKCt8Nw18qOod6TaAUL7tfm8pUbCPKShEN0jJOS/pubchart?oid=2020451260&format=interactive",
+  sheets_gid="1005292860",
+  sql_file="adoption.sql"
+) }}
+
+Let's start with understanding which SSG is most popular. Nuxt.js covers 52.6% of Jamstack sites. Next.js is in second place with 36.8%, third one is Gatsby with 6.7%, followed by Hugo at 2.5%.
+
+{# TODO - add something more, rather than just repeat the description #}
```

Comment:
@denar90 my general feedback is a lot of the commentary on the charts are just repeating the stats. Have you anything more to say on what the results mean?

Reply:
updated